Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,22 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python?(python)
+        return true if build.with?(python)
+        (requirements.to_a | deps).any? { |r| r.name == python && r.required? }
+      end
+
       # Helper method for the common case of installing a Python application.
       # Creates a virtualenv in `libexec`, installs all `resource`s defined
       # on the formula, and then installs the formula.
-      def virtualenv_install_with_resources
-        venv = virtualenv_create(libexec)
+      def virtualenv_install_with_resources(options = {})
```

Comment:
`virtualenv_install_with_resources :using => "python3"` actually reads quite nicely to me!